packages/chain-adapters/src/cosmossdk/CosmosSdkBaseAdapter.ts

```diff
@@ -90,9 +91,44 @@ export abstract class CosmosSdkBaseAdapter<T extends CosmosChainTypes> implement
     input: chainAdapters.TxHistoryInput
   ): Promise<chainAdapters.TxHistoryResponse<T>> {
     try {
-      const { data } = await this.providers.http.getTxHistory({ pubkey: input.pubkey })
-      console.warn(data)
-      throw new Error('Method not implemented.')
+      const { data } = await this.providers.http.getTxHistory({
+        pubkey: input.pubkey,
+        pageSize: input.pageSize,
+        cursor: input.cursor
+      })
+
+      const txs = await Promise.all(
+        data.txs.map(async (tx) => {
+          const parsedTx = await this.parser.parse(tx, input.pubkey)
+
+          return {
+            address: input.pubkey,
+            blockHash: parsedTx.blockHash,
+            blockHeight: parsedTx.blockHeight,
+            blockTime: parsedTx.blockTime,
+            caip2: this.getCaip2(),
+            chain: this.getType(),
+            confirmations: parsedTx.confirmations,
+            txid: tx.txid,
```

Comment:
might as well use `parsedTx` data throughout.
```suggestion
            txid: parsedTx.txid,
```